File: SuperiorPackGroup/BLL/FinishedProductAvailabilityReportBLL.vb
Imports SuperiorPackGroup.SPGReportsTableAdapters

Public Class FinishedProductAvailabilityReportBLL

    Private m_FinishedProductAvailabilityReportTableAdapter As FinishedProductAvailabilityReportTableAdapter = Nothing

    Public ReadOnly Property Adapter() As FinishedProductAvailabilityReportTableAdapter
        Get
            If (Me.m_FinishedProductAvailabilityReportTableAdapter Is Nothing) Then
                Me.m_FinishedProductAvailabilityReportTableAdapter = New FinishedProductAvailabilityReportTableAdapter
                Me.m_FinishedProductAvailabilityReportTableAdapter.Connection.ConnectionString = My.Settings.UserConnectionString
            End If
            Return Me.m_FinishedProductAvailabilityReportTableAdapter
        End Get
    End Property

    Public Function GetFinishedProductAvailabilityReport(ByVal customerID As Nullable(Of Integer), ByVal inactiveItems As Boolean?, ByVal inactiveCustomers As Boolean?, _
                                                         ByVal items As String) As SPGReports.FinishedProductAvailabilityReportDataTable

        If inactiveCustomers.Value = True Then
            inactiveCustomers = Nothing
        End If

        If inactiveItems.Value = True Then
            inactiveItems = Nothing
        End If

        If String.IsNullOrEmpty(items) Then
            Return Adapter.GetFinishedProductAvailabilityReport(customerID, inactiveCustomers, inactiveItems)
        Else
            Return Adapter.GetFinishedProductAvailabilityByItem(customerID, inactiveCustomers, items)
        End If

    End Function

End Class

File: SuperiorPackGroup/BLL/FinishedProductAvailabilityReportBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using SuperiorPackGroup.SPGReportsTableAdapters;

namespace SuperiorPackGroup
{
	public class FinishedProductAvailabilityReportBLL
	{

		private FinishedProductAvailabilityReportTableAdapter m_FinishedProductAvailabilityReportTableAdapter = null;

		public FinishedProductAvailabilityReportTableAdapter Adapter
		{
			get
			{
				if (this.m_FinishedProductAvailabilityReportTableAdapter == null)
				{
					this.m_FinishedProductAvailabilityReportTableAdapter = new FinishedProductAvailabilityReportTableAdapter();
					this.m_FinishedProductAvailabilityReportTableAdapter.Connection.ConnectionString = Properties.Settings.Default.UserConnectionString;
				}
				return this.m_FinishedProductAvailabilityReportTableAdapter;
			}
		}

		public SPGReports.FinishedProductAvailabilityReportDataTable GetFinishedProductAvailabilityReport(int? customerID, bool? inactiveItems, bool? inactiveCustomers, string items)
		{

			if (inactiveCustomers.Value == true)
			{
				inactiveCustomers = null;
			}

			if (inactiveItems.Value == true)
			{
				inactiveItems = null;
			}

			if (string.IsNullOrEmpty(items))
			{
				return Adapter.GetFinishedProductAvailabilityReport(customerID, inactiveCustomers, inactiveItems);
			}
			else
			{
				return Adapter.GetFinishedProductAvailabilityByItem(customerID, inactiveCustomers, items);
			}

		}

	}

}